### PR TITLE
[monorepo] fix: update latest test image to use python 3.12

### PR DESCRIPTION
**Pull request by Wayonb**
## What is the current behavior?
Tests are currently not run with Python 3.12

## What's the issue?
Python 3.12 is now the default version on some Linux distributions.

## How have you changed the behavior?
Use Python 3.12 for testing in the latest image.
Move to Pylint 3, which supports Python 3.12
Remove Pylint-Quotes since does not support Pylint 3

## How was this change tested?
Will run tests in Jenkins.

### CODE REVIEW

File: jenkins/docker/ubuntu/javascript.Dockerfile [hunk 1]
```diff
@@ -17,11 +17,11 @@ RUN apt-get install -y wget gnupg \
 
 # nodejs
 ENV NODE_OPTIONS="--dns-result-order=ipv4first"
-ARG NODE_MAJOR=18
+ARG NODEJS_VERSION=18
```
Comment (Jaguar0625) [inline jenkins/docker/ubuntu/javascript.Dockerfile:20]: fyi - 18 is in maintenance, maybe we should look into upgrading to active (20) in separate PR

File: jenkins/docker/windows/javascript.Dockerfile [hunk 2]
```diff
@@ -12,9 +12,7 @@ RUN Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser; `
 	$command='c:\scoop.ps1 -RunAsAdmin'; `
 	iex $command; `
 	del c:\scoop.ps1; `
-	scoop install git shellcheck nodejs-lts cygwin rustup; `
-	scoop bucket add versions; `
-	scoop install versions/python311; `
+	scoop install git shellcheck nodejs-lts cygwin rustup python3; `
```
Comment (Jaguar0625) [inline jenkins/docker/windows/javascript.Dockerfile:15]: what python version is that? 3.11?

File: linters/python/.pylintrc [hunk 3]
```diff
@@ -276,7 +276,7 @@ missing-member-max-choices=1
 
 # List of additional names supposed to be defined in builtins. Remember that
 # you should avoid to define new builtins when possible.
-additional-builtins=_
+additional-builtins=
```
Comment (Jaguar0625) [inline linters/python/.pylintrc:279]: why this change?

**Review by Jaguar0625 — APPROVED** [review]

File: jenkins/docker/ubuntu/javascript.Dockerfile (hunk 1 above)
Comment (Wayonb) [reply jenkins/docker/ubuntu/javascript.Dockerfile:20]: Yeah, I will also update the default value.  This will be overwritten by values in the ``baseImages.yaml``

File: jenkins/docker/windows/javascript.Dockerfile (hunk 2 above)
Comment (Wayonb) [reply jenkins/docker/windows/javascript.Dockerfile:15]: Scoop will always install the latest Python version on Windows.  Currently, this is Python 3.12.1.

This is an excellent way to find out when our Python SDK is broken with the latest release 😅

File: linters/python/.pylintrc (hunk 3 above)
Comment (Wayonb) [reply linters/python/.pylintrc:279]: This is a mistake, will revert 👍🏾

**Review by Jaguar0625 — APPROVED** [review]
